File: Spring/balance-checker-api/src/main/java/io/naav97/balance_checker_api/models/Transaction.java
package io.naav97.balance_checker_api.models;
import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

@Entity
public class Transaction {

  private @Id @GeneratedValue Long id;
  private String userFrom;
  private String userTo;
  private double amount;
  private LocalDateTime time;

  public String get_user_from() {
    return this.userFrom;
  }

  public String get_user_to() {
    return this.userTo;
  }

  public double get_amount() {
    return this.amount;
  }

  public LocalDateTime get_time() {
    return this.time;
  }

  public void set_user_from(String p_uid) {
    this.userFrom = p_uid;
  }

  public void set_user_to(String p_uid) {
    this.userTo = p_uid;
  }

  public void set_amount(double p_amo) {
    this.amount = p_amo;
  }

  public void set_time(LocalDateTime p_time) {
    this.time = p_time;
  }
}

File: Gin/balance-chacker-api/models/transaction.go
package models

import (
  "time"
)

type Transaction struct {
  Id int `json:"id"`
  UserFrom string `json:"user_from"`
  UserTo string `json:"user_to"`
  Amount float64 `json:"amount"`
  Time time.Time `json:"time"`
}

var Transactions = []Transaction{}

func AddTrans(trans Transaction) {
  Transactions = append(Transactions, trans)
}

func GetUserTrans(un string) []Transaction {
  var re []Transaction
  for _, t := range Transactions {
    if t.UserFrom == un || t.UserTo == un {
      re = append(re, t)
    }
  }
  return re
}
